profiling/README.md

```diff
@@ -1,17 +1,50 @@
+# PHP Profiler
+
 The profiler is implemented in Rust. To see the currently required Rust
 version, refer to the [rust-toolchain](rust-toolchain) file. The profiler
-requires PHP 7.1+, and does not support debug nor ZTS builds. There are bits
-of ZTS support in the build system and profiler, but it's not complete.
+requires PHP 7.1+, and does not support debug nor ZTS builds. There are bits of
+ZTS support in the build system and profiler, but it's not complete.
+
+## Compiling
 
 The command `cargo build` will run the [build.rs](build.rs) script, which is
 how it adapts to various PHP versions. The
 [bindgen](https://crates.io/crates/bindgen) crate is used to generate Rust
 bindings to the Zend Engine. Although bindgen is pretty good, there are things
-like complex macro expansions which it doesn't understand, so there is a bit
-of C code to do things that Rust/bindgen isn't good at.
-
-The [src/bindings.rs](src/bindings.rs) file manually defines certain structs
-and function definitions instead of letting bindgen handle it. This allows us
-to gloss over minor differences in const-correctness in the engine definitions
-across versions, as well as provide more idiomatic types in some cases where
-they are ABI compatible.
+like complex macro expansions which it doesn't understand, so there is a bit of
+C code to do things that Rust/bindgen isn't good at.
+
+The [src/bindings/mod.rs](src/bindings/mod.rs) file manually defines certain
+structs and function definitions instead of letting bindgen handle it. This
+allows us to gloss over minor differences in const-correctness in the engine
+definitions across versions, as well as provide more idiomatic types in some
+cases where they are ABI compatible.
+
+## Testing
+
+The command `cargo test` will run the tests on the profiler.
+
+To see if the profiler is recognised by your PHP version as an extension you
+may run `php -d extension=target/debug/libdatadog_php_profiling.so --ri
+datadog-profiling` and check the output.
+
+The following commands will help you run the PHPT tests from
+[tests/phpt/](tests/phpt):
+
+```sh
+cp -v "$(find $(php-config --prefix) -name run-tests.php)" .
+export TEST_PHP_EXECUTABLE=$(which php)
+php run-tests.php -d extension=target/release/libdatadog_php_profiling.dylib tests/phpt
```

Comment:
.dylib .so mismatch, stick to one - probably the thing that is helpful for most people is using .so, but you could note somewhere that SHLIB_SUFFIX is different for mac (although these users are probably aware).

Reply:
Nice find with the `.dylib` 👍

Reply:
I was not aware of the `.dylib` file ending on mac, but I also was not a mac user two weeks ago 😉 
I would also think, that most people have the `.so` extension and even if not, it is pretty easy to find the file

Reply:
I am not aware of the nuances of `.so` and `.dylib` on Mac, but I do know:

1. `cargo` builds a `.dylib` right now.
2. Renaming this to `datadog-profiling.so` when copying it into the `php-config --extension-dir` is fine, and it will load just fine by PHP when doing `-dextension=datadog-profiling`.

So... `php run-tests.php -d extension=target/release/libdatadog_php_profiling.dylib tests/phpt` is correct... at least for now.

Reply:
Found in the [Rust Linkage Docs](https://doc.rust-lang.org/reference/linkage.html) that for `cdylib` crates Rust "will create *.so files on Linux, *.dylib files on macOS, and *.dll files on Windows.".

I think it is safe to assume most people will use this together with Linux, so going with the `.so` file ending should be okay from my point of view.
CircleCI running on Linux also creates `.so` files: https://github.com/DataDog/dd-trace-php/blob/master/.circleci/config.yml#L1698

We could still add a hint for Mac users, that the file ending most likely will be `.dylib` in their case.